pkg/test/testreconciler/simpletest/controller_test.go

```diff
@@ -138,7 +138,7 @@ func testSimpleReconciler(h *testharness.Harness, testdir string, applier applie
 
 	mgrContext, mgrStop := context.WithCancel(ctx)
 	go func() {
-		time.Sleep(2 * time.Second)
+		time.Sleep(1 * time.Second)
```

Comment:
With 2 second reconcile window, the test always ends up with several more roundtrips and this results as more `GET ..configmap, namespace` `PATCH configmap namespace`. Since all HTTP requests are logged and compared by golden test, the e2e test always fails with 3000+ lines added to the simpletest/testdata/reconcile/ssa/create/expected-http.yaml, than expected. 

Reducing the time window is a tmp workaround. Another option we can do is to skip logging the http request for "GET configmap/namespace", and "PATCH if no change"  (current e2e test logs the PATCH even if the manifest is not changed. https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/mockkubeapiserver/patchresource.go#L122-L124

Reply:
Ideally we would "know" that reconciliation is done, but that's actually a bit tricky to know.  A longer window ensures that there's no ongoing reconciliation, a shorter window means the tests run faster.

The "real" fix here I think is to ensure that deployments go ready, which I think is why we are looping here.  We're doing that separately.  I think reducing the time is fine here, we can revisit in future once all the in-flight PRs have landed.